test/FunctionalTests/Microsoft.OData.Core.Tests/ValidationHelper.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.OData.Tests
+{
+    public class ValidationHelper
+    {
+        /// <summary>
+        /// This method takes 2 generic types and see if one is the deep copy  of other
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="obj"></param>
+        /// <param name="copy"></param>
+        /// <returns></returns>
+        public static List<string> GetDifferences<T>(T obj, T copy)
+        {
+            var diff = new List<string>();
+            EvaluateDifferences(obj, copy, diff);
+
+            return diff;
+        }
+
+        /// <summary>
+        /// Generic Method to evaluate differences of 2 objects, caled recursively
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="obj"></param>
+        /// <param name="copy"></param>
+        /// <param name="diff"></param>
+        private static void EvaluateDifferences<T>(T obj, T copy, List<string> diff)
+        {
+
+            foreach (var prop in obj.GetType().GetProperties().Where(x => x.CanRead))

```

Comment:
verify the input and remove unused the line

Reply:
Resolved